llama_index/vector_stores/redis.py

```diff
@@ -299,7 +299,7 @@ def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResul
                         NodeRelationship.SOURCE: RelatedNodeInfo(node_id=doc.doc_id)
                     },
                 )
-            ids.append(doc.id)
+            ids.append(doc.id.replace(self._prefix + "_", ""))
```

Comment:
is the prefix added intentionally ?

Reply:
The prefix is added to avoid conflicts in the Redis database when using Redis as IndexStore, VectorStore and Docstore.

Reply:
The problem is that if the prefix is not removed the documents can't be retrieved because they don't have it that prefix when saved in the Docstore.